src/server/server_family.cc

```diff
@@ -118,19 +122,22 @@ string InferLoadFile(fs::path data_dir) {
   if (fs::exists(fl_path))
     return fl_path.generic_string();
 
+  SubstituteFilenameTsPlaceholder(&fl_path, "*");
   if (!fl_path.has_extension()) {
-    std::string glob = absl::StrCat(fl_path.generic_string(), "*");
-    io::Result<io::StatShortVec> short_vec = io::StatFiles(glob);
-
-    if (short_vec) {
-      auto it = std::find_if(short_vec->rbegin(), short_vec->rend(), [](const auto& stat) {
-        return absl::EndsWith(stat.name, ".rdb") || absl::EndsWith(stat.name, "summary.dfs");
-      });
-      if (it != short_vec->rend())
-        return it->name;
-    } else {
-      LOG(WARNING) << "Could not stat " << glob << ", error " << short_vec.error().message();
-    }
+    fl_path += ".*";
+  }
+  io::Result<io::StatShortVec> short_vec = io::StatFiles(fl_path.generic_string());
+
+  if (short_vec) {
+    // TODO: For the case of files with timestamps, we should sort the results
+    // and select the newest dump.
+    auto it = std::find_if(short_vec->rbegin(), short_vec->rend(), [](const auto& stat) {
+      return absl::EndsWith(stat.name, ".rdb") || absl::EndsWith(stat.name, "summary.dfs");
+    });
```

Comment:
`io::StatFiles` uses glob underneath that should sort files by default https://man7.org/linux/man-pages/man3/glob.3.html,

this is why we use reverse iterators in find_if

Reply:
Thanks! :) I replaced the TODO with a comment.